plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3366,4 +3368,19 @@ private static Optional<String> getQueryId(Table table)
     {
         return Optional.ofNullable(table.getParameters().get(PRESTO_QUERY_ID_NAME));
     }
+
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
```

Comment:
nit: Please move this method before `applyTableScanRedirect` to ensure overloaded methods order.